### PR TITLE
Refactor manifests/avoid duplication

### PR DESCRIPTION
**Pull request by igoropaniuk**
1. Use common repositories defined in common.xml manifest to avoid
    code duplication.
2. Drop manifest for Verdin iMX8M Mini, as it's not maintained anymore.

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
I'd rather wait with an eventual refactoring of the manifest until after the release.

**Comment by jforissier** [comment]
> I'd rather wait with an eventual refactoring of the manifest until after the release.

Agreed. As usual, and since -rc1 has been tagged already, I will only merge fixes.

**Comment by jbech-linaro** [comment]
I'm OK with these changes and it's a nice clean up. But at the same time, if we for any reason in the future would need to move something to a specific git commit for a particular device, then we cannot do that as easy any longer. For that you'd have to use a local manifest, that works locally. However, historically we haven't had that happen a lot, so again, I'm OK with this, so please use my RB tag.

`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
You're supposed to be able to override what's in a previously included file. See for example `fvp-ts.xml`.

**Comment by jbech-linaro** [comment]
> You're supposed to be able to override what's in a previously included file. See for example `fvp-ts.xml`.

OK, then I have no concerns. Thanks!

File: am43xx.xml [hunk 1]
```diff
@@ -5,19 +5,12 @@
 
         <default remote="github" revision="master" />
```
Comment (jenswi-linaro) [inline am43xx.xml:6]: Could this be taken from `common.xml` instead?

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commits
"verdin: drop manifest" and
"include: move common repositories to separate file".

For commit "common: use common manifest to avoid code duplication":
commit message could mention that reference to optee_benchmark repository is removed as the feature is also currently being removed. Or you could have a dedicated commit for removing these refs.

**Comment by etienne-lms** [comment]
My bad, optee_benchmark is listed in _common.xml_.

`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commit
"common: use common manifest to avoid code duplication".

**Comment by igoropaniuk** [comment]
@etienne-lms I'll drop optee_manifest in a separate PR

File: am43xx.xml (hunk 1 above)
Comment (igoropaniuk) [reply am43xx.xml]: fixed

**Comment by igoropaniuk** [comment]
Addressed @jenswi-linaro comment, applied tags from @jbech-linaro @etienne-lms , rebased

File: common.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        <project path="optee_os"             name="OP-TEE/optee_os.git" />
+        <project path="optee_test"           name="OP-TEE/optee_test.git" />
+
+        <!-- linaro-swg gits -->
+        <project path="linux"                name="linaro-swg/linux.git"                  revision="optee" clone-depth="1" />
+        <project path="optee_benchmark"      name="linaro-swg/optee_benchmark.git"/>
+        <project path="optee_examples"       name="linaro-swg/optee_examples.git" />
+
+        <!-- buildroot git -->
+        <project path="buildroot"            name="buildroot/buildroot.git"               revision="refs/tags/2022.11.1" clone-depth="1" remote="github" />
```
Comment (jenswi-linaro) [inline common.xml:17]: The default remote is "github" already.

File: versal.xml [hunk 3]
```diff
@@ -1,20 +1,11 @@
...
-	<!-- OP-TEE gits -->
-	<project path="optee_os"		name="OP-TEE/optee_os.git" />
-	<project path="optee_client"		name="OP-TEE/optee_client.git" />
-	<project path="optee_docs"		name="OP-TEE/optee_docs.git" />
-	<project path="optee_test"		name="OP-TEE/optee_test.git" />
-	<project path="build"			name="OP-TEE/build.git" >
-		<linkfile src="versal.mk" dest="build/Makefile" />
+	<include name="common.xml" />
+	<project path="build"                   name="OP-TEE/build.git" >
+			<linkfile src="qemu_v8.mk" dest="build/Makefile" />
```
Comment (jenswi-linaro) [inline versal.xml:5]: s/qemu_v8.mk/versal.mk/ ?

File: common.xml (hunk 2 above)
Comment (igoropaniuk) [reply common.xml]: fixed, thanks!

File: versal.xml (hunk 3 above)
Comment (igoropaniuk) [reply versal.xml]: fixed

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by igoropaniuk** [comment]
@jenswi-linaro tag applied. I've just stumbled upon another issue, rpi3 was using `optee` branch instead of `rpi3-optee-6.7`, this is fixed already

**Pull request merged**